api/src/database/get-ast-from-query/lib/parse-fields.ts

```diff
@@ -191,11 +191,15 @@ export async function parseFields(
 			};
 
 			for (const relatedCollection of allowedCollections) {
+				const existingFields = Array.isArray(nestedFields)
+					? nestedFields.filter((field) => field in (context.schema.collections[relatedCollection]?.fields ?? {}))
+					: [];
+
```

Comment:
I think we should support the wildcard case, eg: `fields=m2a.items.*`.

```suggestion
				let existingFields: string[] = [];

				if (Array.isArray(nestedFields)) {
					if (nestedFields.includes('*')) {
						existingFields = Object.keys(context.schema.collections[relatedCollection]?.fields ?? {});
					} else {
						existingFields = nestedFields.filter(
							(field) => field in (context.schema.collections[relatedCollection]?.fields ?? {}),
						);
					}
				}

```

Reply:
Very good catch! But I think we can just populate nested fields with '*' instead of expanding them here, since expandWildcards takes care of that

Reply:
I'll commit it like that and give it test run once I'm back at the desk.